### PR TITLE
Auth Code Login

### PR DESCRIPTION
**Pull request by BottleRocket-Colin**
Not the prettiest but it works.  Also left some old token refresh logic in place.  Hopefully the token will be long lived enough where this doesn't cause any issues.

### CODE REVIEW

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/BaseFragment.kt [hunk 1]
```diff
@@ -36,4 +40,20 @@ abstract class BaseFragment<FRAGMENT_VIEW_MODEL : BaseViewModel> : Fragment() {
     protected open fun setupNavigationObservers() {
         fragmentViewModel.observeNavigationEvents(this)
     }
+
+    // /////////////////////////////////////////////////////////////////////////
+    // Compose Helpers
+    // /////////////////////////////////////////////////////////////////////////
+    @Suppress("FunctionName")
+    fun ComposeScreen(content: @Composable () -> Unit) =
```
Comment (BottleRocket-Colin) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/BaseFragment.kt:48]: New helper function to reduce boilerplate in fragment.

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+@SuppressLint("SetJavaScriptEnabled")
+@Composable
+fun RequestAuth(url: String, onAuthCode: (String) -> Unit) {
+    AndroidView(factory = {
+        WebView(it).apply {
+            webViewClient = object : WebViewClient() {
+                override fun shouldOverrideUrlLoading(view: WebView?, request: WebResourceRequest?): Boolean {
+                    //  This has to match callback URL defined for bitbucket api key.
+                    if (request?.url?.toString()?.contains("www.bottlerocketstudios.com") == true) {
```
Comment (BottleRocket-Colin) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt:69]: This host must match the one used for BitBucket API key callback.

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        Image(
+            painter = painterResource(id = R.drawable.ic_logo),
+            contentDescription = stringResource(id = R.string.splash_image_content_description)
+        )
+
+        Text(
+            modifier = Modifier
+                .padding(top = Dimens.grid_2)
+                .padding(start = Dimens.grid_2, end = Dimens.grid_2),
+            text = "BitBucket now uses an auth code to login.",
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt:107]: nit: Add these to string resources in future cleanup PR

File: compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            textAlign = TextAlign.Center,
+            style = MaterialTheme.typography.h2,
+        )
+
+        Text(
+            modifier = Modifier
+                .padding(top = Dimens.grid_2)
+                .padding(start = Dimens.grid_2, end = Dimens.grid_2),
+            text = "Tapping Login below will take you to a web view to authenticate",
```
Comment (jaredstockton) [inline compose/src/main/java/com/bottlerocketstudios/compose/auth/AuthCode.kt:116]: nit: Add this to string resources in future cleanup PR

File: app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/auth/AuthCodeViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+    // UI
+    val requestUrl = MutableStateFlow("")
+    val devOptionsEnabled = buildConfigProvider.isDebugOrInternalBuild
+
+    // /////////////////////////////////////////////////////////////////////////
+    // Callbacks
+    // /////////////////////////////////////////////////////////////////////////
+    fun onLoginClicked() {
+        requestUrl.value = "https://bitbucket.org/site/oauth2/authorize?client_id=$BITBUCKET_KEY&response_type=code"
```
Comment (jaredstockton) [inline app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/auth/AuthCodeViewModel.kt:40]: We might need to bring `BITBUCKET_KEY` (and possibly `BITBUCKET_SECRET`) into `BuildConfigProvider` if we're unable to unit test this function in a future PR

**Review by jaredstockton — APPROVED** [review]
Looks good overall! Some things can be tweaked in future work as getting auth working again is the primary thing to address right now

**Pull request merged**